### PR TITLE
XY Chart: Support time and time range selection

### LINKED ISSUE #121: XY Chart: support time and time range selection
timegraph chart supports selecting a time or time range, but not the XY chart. This support should be added

### PR DESCRIPTION
**Pull request by arfio**
When clicking on the XY chart, the user can select a time.
When clicking and dragging, the user can select a time range.

fixes #121 

Signed-off-by: Arnaud Fiorini <fiorini.arnaud@gmail.com>

### CODE REVIEW

**Review by tahini — CHANGES_REQUESTED** [review]
Thanks for this feature! It's a great improvement and it works well! ... as long as the selection is in range... See the screencast below, when I move the time range so that the selection is out of sight, the xy chart is completely selected.

xySelection

File: packages/react-components/src/components/xy-output-component.tsx [hunk 1]
```diff
@@ -154,28 +181,41 @@ export class XYOutputComponent extends AbstractTreeOutputComponent<AbstractOutpu
...
+                return;
+            }
+            const minValue = this.findNearestValue(min, ticks);
+            const minPixel = xScale.getPixelForValue(min, minValue);
+            const maxValue = this.findNearestValue(max, ticks);
+            let maxPixel = xScale.getPixelForValue(max, maxValue);
+            // In the case the selection is going out of bounds, the pixelValue needs to be in the displayed range.
+            if (maxPixel === 0) {
+                maxPixel = chart.chartArea.right;
+            }
```
Comment (tahini) [inline packages/react-components/src/components/xy-output-component.tsx:197]: Just asking, why do you have to change this method? The behavior of drawing the the selection in XY chart is correct in master. You added the possibility to also select from the XY chart. I'm just wondering why this needs to be modified too.

**Review by tahini — APPROVED** [review]
It works great! Thanks!

File: packages/react-components/src/components/xy-output-component.tsx (hunk 1 above)
Comment (arfio) [reply packages/react-components/src/components/xy-output-component.tsx:197]: if your selection is partially out of range, the selection is not correctly displayed. Also the borders shown were not the same as for the timeline chart. When I corrected these issues, I had to account for other things as well, start > end and the window out of range completely.

**Review by bhufmann — APPROVED** [review]
Looks good to me! Thanks for the improvement

**Review by bhufmann — APPROVED** [review]

**Pull request merged**